### DISCUSSION TITLE
How can I do code format before I send a PR ？

### DISCUSSION THREAD

**Discussion by knightXun** [openxla/xla#16335] [Q&A] [answered]
How can I do code format before I send a PR ？

**Comment by beckerhe** [comment, marked as answer]
You can (and should) use `clang-format` for code formatting. It's available as a package on basically all Linux distributions.

I recommend using the Git-integration which only formats the lines of code that you have changed.

`git clang-format` formats all the changes that have not yet been commited.

If you have already committed your changes, then `git clang-format HEAD~1` also formats the changes from the last commit.

**Reply by knightXun** [reply]
I notice XLA code-format ci always fail, and the format-ci logs show these errors are not from my pr. What should I do in this situation?

**Reply by beckerhe** [reply]
There seems to be a disagreement between different versions of clang-format about what's the correct formatting for `algebraic_simplifier_test.cc`. Please mention in the PR description that the suggested formatting changes are unrelated to your PR
